src/test/java/co/rsk/federate/adapter/ThinConverterTest.java

```diff
@@ -4,15 +4,58 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNull;
 
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.peg.constants.BridgeConstants;
+import co.rsk.peg.constants.BridgeMainNetConstants;
 import co.rsk.peg.constants.BridgeRegTestConstants;
 import java.math.BigInteger;
 import java.util.ArrayList;
+import java.util.stream.Stream;
+import org.bitcoinj.core.ECKey;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 class ThinConverterTest {
 
+    private static final BigInteger NEGATIVE_CHAIN_WORK = BigInteger.valueOf(-1);
+    private static final BigInteger EIGHT_BYTES_WORK_V1 = new BigInteger("ffffffffffffffff", 16); // 8 bytes
+    private static final BigInteger MAX_WORK_V1 = new BigInteger(/* 12 bytes */ "ffffffffffffffffffffffff", 16);
+    private static final BigInteger MAX_WORK_V2 = new BigInteger(/* 32 bytes */
+        "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", 16);
+    private static final BigInteger TOO_LARGE_WORK_V1 = new BigInteger(/* 13 bytes */ "ffffffffffffffffffffffffff", 16);
+    private static final BigInteger TOO_LARGE_WORK_V2 = new BigInteger(/* 33 bytes */
+        "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", 16);
+
+    private static final int height = 200;
     private static int nhash = 0;
 
+    private static final BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+    org.bitcoinj.core.NetworkParameters bitcoinCoreParams = org.bitcoinj.params.MainNetParams.get();
+    co.rsk.bitcoinj.core.NetworkParameters bitcoinjThinParams = co.rsk.bitcoinj.params.MainNetParams.get();
+
+    private static final ECKey userKey = ECKey.fromPrivate(BigInteger.valueOf(100));
+
+    public static Stream<Arguments> validChainWorkArgsProvider() {
```

Comment:
```suggestion
    public static Stream<Arguments> validChainWorkV2ArgsProvider() {
```

Reply:
No. This includes both formats of chainwork.

Reply:
how is the too large work v1 valid in v1?